src/lib/work_partitioner/work_partitioner.ml

```diff
@@ -1 +1,242 @@
+open Core_kernel
+
+open struct
+  module Work = Snark_work_lib
+end
+
 module Snark_worker_shared = Snark_worker_shared
+module Zkapp_command_job_pool =
+  Job_pool.Make (Work.Id.Single) (Pending_zkapp_command)
+module Sent_zkapp_job_pool =
+  Job_pool.Make (Work.Id.Sub_zkapp) (Work.Spec.Sub_zkapp.Stable.Latest)
+module Sent_single_job_pool =
+  Job_pool.Make (Work.Id.Single) (Work.Spec.Single.Stable.Latest)
+
+(* TODO: use proper cache DB later *)
+let _proof_cache_db = Proof_cache_tag.create_identity_db ()
+
+(* WARN: we're mixing ID for `pairing_pool` and `zkapp_command_jobs. *)
```

Comment:
should we create two id generators? I know, there is no real value, but just for keeping the two completely separate, as extra caution?